MapboxNavigation/NavigationMapView.swift

```diff
@@ -483,15 +491,18 @@ open class NavigationMapView: MGLMapView, UIGestureRecognizerDelegate {
             // In case of main route there is the ability to provide custom `MGLShape` for either route or route casing by implemeting
             // `NavigationMapViewDelegate.navigationMapView(_:shapeFor:)` or `NavigationMapViewDelegate.navigationMapView(_:simplifiedShapeFor:)`.
             if index == 0 {
+                
                 let routeShape = navigationMapViewDelegate?.navigationMapView(self, shapeFor: [route]) ??
                     shape(for: route, legIndex: legIndex, isAlternateRoute: false)
                 
                 let routeSource = addRouteSource(style, identifier: routeSourceIdentifier, shape: routeShape)
                 
+                let fractionTraveledForGradient = routeLineTracksTraversal ? fractionTraveled : 0.0
```

Comment:
Looks like this check will not help to prevent shift at the start of navigation because `fractionTraveled` value is updated only after showing route line itself. I understand that this might be a tricky case to fix, so please create separate ticket for this, we can tackle this as a tail-work.

Reply:
I can see that you've already created ticket in https://github.com/mapbox/mapbox-navigation-ios/issues/2757. Thanks!